.buildkite/scripts/lifecycle/comment_on_pr.ts

```diff
@@ -51,19 +59,26 @@ export function commentOnPR({
   });
 
   if (context) {
-    return upsertComment({ commentBody: message, commentContext: context, clearPrevious });
+    return upsertComment(
+      { commentBody: message, commentContext: context, clearPrevious },
+      repositoryOwner,
+      repositoryBase,
+      issueNumber
+    );
   } else {
-    return addComment(message);
+    return addComment(message, repositoryOwner, repositoryBase, issueNumber);
   }
 }
 
 if (require.main === module) {
-  const args = parseArgs<{
-    context?: string;
-    message: string;
-    'clear-previous'?: boolean | string;
-  }>(process.argv.slice(2), {
-    string: ['message', 'context'],
+  const args = parseArgs<
+    Omit<CommentOnPRArgs, 'clearPrevious' | 'issueNumber'> & {
```

Comment:
this bit is quite hard to parse for a human. Maybe it's easier to go with an explicit type definition, instead of typelevel operations?

Reply:
I simplified the implementation